src/schema/ecommerce/set_order_payment_mutation.ts

```diff
@@ -91,9 +91,9 @@ export const SetOrderPaymentMutation = mutationWithClientMutationId({
       creditCardId,
     }).then(result => {
       if (result.errors) {
-        throw Error(result.errors.map(d => d.message))
+        throw Error(result.errors.map(d => d.message).join("\n"))
       }
-      const { order, errors } = result.data.ecommerce_setPayment
+      const { order, errors } = result.data!.ecommerce_setPayment
```

Comment:
Yup, it is TS syntax that asserts the value cannot be `null`/`undefined`. We can safely do that because `result.errors` is checked for above, but there’s no way for TS to know that that check means this is safe. Alternatively we could have added an extra `if (result.data) { … }` check and TS would have narrowed the type to remove `null`/`undefined` as well, but it felt like that runtime overhead isn’t necessary when I know better than the compiler 🤷‍♂️